components/pages/workPageLayout/WorkPageHeader.tsx

```diff
@@ -51,12 +71,30 @@ const WorkPageHeader = ({ work }: WorkPageHeaderProps) => {
     undefined
   )
   const findInitialSliderValue = () => {
+    const searchParams = new URLSearchParams(window.location.search)
+    // If we have a material type specified in the URL, we use that
+    if (
+      searchParams.has("type") &&
+      slideSelectOptions.some(option => option.render === searchParams.get("type"))
+    ) {
+      return slideSelectOptions.find(option => option.render === searchParams.get("type"))
+    }
     return slideSelectOptions.find(option => {
       return selectedManifestation?.materialTypes.find(materialType => {
         return materialType.materialTypeGeneral.code.includes(option.value)
       })
     })
   }
+  const languageIsoCode = getManifestationLanguageIsoCode(selectedManifestation)
+  const onOptionSelect = (optionSelected: SlideSelectOption) => {
+    setSelectedManifestation(
+      getManifestationByMaterialType(work, optionSelected.value) ||
+        work.manifestations.bestRepresentation
+    )
+    const searchParams = new URLSearchParams(window.location.search)
+    searchParams.set("type", optionSelected.render)
+    router.push(`${window.location.pathname}?${searchParams.toString()}`)
+  }
```

Comment:
Instead of setting the selected manifestation here. I would suggest setting the url param and make a useeffect that listens to changes on url param "type". Inside that useEffect set the new selected manifestation type. :) 

On route push function consider adding the following to in options
`{
      scroll: false
 }`
This will prevent scrolling to the top of the page when changing type.
https://nextjs.org/docs/app/api-reference/components/link#reference

Reply:
It turned out to be quite an undertaking haha. But I did it <3

Reply:
Great! Looks good. Please add the icons to the switch or create a ticket on the issue. 

<img width="770" alt="Screenshot 2024-12-05 at 14 49 26" src="https://github.com/user-attachments/assets/3a2016cb-1394-4ba8-8cd5-96ffdfbfa603">